### DISCUSSION TITLE
Introduce/extend manufacturing line quickstart with job dependencies

### DISCUSSION THREAD

**Discussion by mcimbora** [TimefoldAI/timefold-solver#385] [Ideas]
Let's say we've got a model similar to this:

```
ManufacturingLine
List<Job> jobs
```

```
Job
List<Job> predecessorJobs
List<Job> successorJobs
```

The jobs have dependencies between each other (Job can start only after the predecessors are finished). At the same time, the manufacturing line can process a single item at time only. This is a fairly common problem, that causes confusion (should I go for food packaging vs project job scheduling). Ideally the docs explain the differences.

Volunteer to implement this (not critical atm).

**Comment by triceo** [comment]
+1. We should have a job shop scheduling quickstart.

**Comment by kentbill** [comment]
Yes, this scene is indeed so practical. If tasks on the same ManufacturingLine have cleaning time (Food Packaging), and the start time of the task has a predecessor task dependency (Project Job Scheduling). I started with the above two examples and tried to implement such model, and always appeared some issues are stuck. 

This is a problem I got when adding predecessor's startTime constraint based on Chained Through Time pattern.
[https://stackoverflow.com/questions/77319671/a-score-corruption-from-chained-through-time-pattern](url)

**Comment by ge0ffrey** [comment]
+1 Let's create an issue to track. Creating one will save us time, instead of costing us time, in the semi long term.

**Comment by triceo** [comment]
Issue created:
https://github.com/TimefoldAI/timefold-quickstarts/issues/141
